databricks/koalas/groupby.py

```diff
@@ -974,10 +975,34 @@ def apply(self, func):
                     return_schema = StructType([StructField("0", return_schema)])
 
         def pandas_groupby_apply(pdf):
+
+            if not is_series_groupby and LooseVersion(pd.__version__) < LooseVersion("0.25"):
+                # `groupby.apply` in pandas<2.5 runs the functions twice for the first group.
+                # https://github.com/pandas-dev/pandas/pull/24748
+                from pandas.core.base import SelectionMixin
+
+                f = SelectionMixin._builtin_table.get(func, func)
+
+                dummy_value = None
```

Comment:
How about naming it like `should_skip_first_call`, and assigning a bool? e.g.:

```python
should_skip_first_call = True

...

def wrapped_func(df):
    nonlocal should_skip_first_call
    if should_skip_first_call:
        should_skip_first_call = False
        if should_return_series:
            return pd.Series()
        else:
            return pd.DataFrame()
    else:
        return f(df)
```

Reply:
sounds great.